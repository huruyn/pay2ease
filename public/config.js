window.API_URL = "https://zippy-retail-api-dev.azurewebsites.net";