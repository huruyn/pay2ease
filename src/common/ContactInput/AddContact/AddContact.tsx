import PropTypes from 'prop-types';
import { Button, FormControl } from 'react-bootstrap';
import {
  Dispatch,
  useState,
} from 'react';
import { useMsal } from '@azure/msal-react';
import Api, { Contact } from '../../../api';

function AddContact({
  onContactCreated,
}: {
  onContactCreated: Dispatch<Contact>;
}) {
  const { instance, accounts } = useMsal();
  const [isProcessing, setIsProcessing] = useState<boolean>(false);
  const [firstName, setFirstName] = useState<string>('');
  const [lastName, setLastName] = useState<string>('');
  const [email, setEmail] = useState<string>('');
  const [nickname, setNickname] = useState<string>('');

  const isValid = (): boolean => {
    const regEmail = /^([a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-zA-Z0-9](?:[a-zA-Z0-9-]*[a-zA-Z0-9])?\.)+[a-zA-Z0-9](?:[a-zA-Z0-9-]*[a-zA-Z0-9])?)$/;

    if (firstName.length < 3) return false;
    if (lastName.length < 3) return false;
    if (!email.match(regEmail)) return false;
    if (nickname && nickname.length < 3) return false;

    return !!firstName && !!lastName && !!email;
  };

  const resetFields = () => {
    setFirstName('');
    setLastName('');
    setEmail('');
    setNickname('');
  };

  const addContact = () => {
    if (isValid()) {
      setIsProcessing(true);
      new Api(instance, accounts[0])
        .postContact({
          firstName,
          lastName,
          email,
          nickname,
        }).then((response: Contact) => {
          resetFields();
          onContactCreated(response);
        }).finally(() => setIsProcessing(false));
    }
  };

  return (
    <>
      <div className="container-header">
        <h2 className="title">Add  Contact</h2>
      </div>
      <div className="container-body">
        <FormControl
          type="text"
          className="common-input"
          placeholder="Firstname"
          name="firstname"
          value={firstName}
          onChange={(evt) => setFirstName(evt.target.value)}
        />
        <FormControl
          type="text"
          className="common-input"
          placeholder="Lastname"
          name="lastname"
          value={lastName}
          onChange={(evt) => setLastName(evt.target.value)}
        />
        <FormControl
          type="text"
          className="common-input"
          placeholder="Email"
          name="email"
          value={email}
          onChange={(evt) => setEmail(evt.target.value)}
        />
        <FormControl
          type="text"
          className="common-input"
          placeholder="Nickname (optional)"
          name="nickname"
          value={nickname}
          onChange={(evt) => setNickname(evt.target.value)}
        />
      </div>
      <div className="action">
        <Button
          className="zippy-btn"
          disabled={isProcessing || !isValid()}
          onClick={() => addContact()}
        >
          {isProcessing && <div className="loading spinner-border" role="status" />}
          Add Contact
        </Button>
      </div>
    </>
  );
}

AddContact.propTypes = {
  onContactCreated: PropTypes.func,
};

AddContact.defaultProps = {
  onContactCreated: undefined,
};

export default AddContact;
