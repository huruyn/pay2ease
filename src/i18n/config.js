import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

i18n.use(initReactI18next).init({
  fallbackLng: 'en-US',
  lng: 'en-US',
  resources: {
    'en-CA': {
      translations: require('./locales/en-CA/translations.json')
    },
    'en-US': {
      translations: require('./locales/en-US/translations.json')
    },
    'es-US': {
      translations: require('./locales/es-US/translations.json')
    },
    'fr-CA': {
      translations: require('./locales/fr-CA/translations.json')
    },
  },
  ns: ['translations'],
  defaultNS: 'translations'
});

i18n.languages = ['en-CA', 'en-US', 'es-US', 'fr-CA'];

export default i18n;