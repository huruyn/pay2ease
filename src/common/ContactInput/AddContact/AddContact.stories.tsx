import { ComponentStory, ComponentMeta } from '@storybook/react';
import '../../../i18n/config';
import '../../../styles/index.scss';
import AddContact from './AddContact';

export default {
  title: 'common/AddContact',
  component: AddContact,
} as ComponentMeta<typeof AddContact>;

const Template: ComponentStory<typeof AddContact> = () => <AddContact />;

export const Default = Template.bind({});
Default.args = {};
