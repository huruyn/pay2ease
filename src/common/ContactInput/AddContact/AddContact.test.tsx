import '../../../i18n/config';
import AddContact from './AddContact';

const ReactTestRenderer = require('react-test-renderer');

describe('AddContact Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(
      <AddContact />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
