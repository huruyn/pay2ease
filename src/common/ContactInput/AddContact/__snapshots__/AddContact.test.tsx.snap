// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddContact Component matches the snapshot 1`] = `
Array [
  <div
    className="container-header"
  >
    <h2
      className="title"
    >
      Add  Contact
    </h2>
  </div>,
  <div
    className="container-body"
  >
    <input
      className="common-input form-control"
      name="firstname"
      onChange={[Function]}
      placeholder="Firstname"
      type="text"
      value=""
    />
    <input
      className="common-input form-control"
      name="lastname"
      onChange={[Function]}
      placeholder="Lastname"
      type="text"
      value=""
    />
    <input
      className="common-input form-control"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value=""
    />
    <input
      className="common-input form-control"
      name="nickname"
      onChange={[Function]}
      placeholder="Nickname (optional)"
      type="text"
      value=""
    />
  </div>,
  <div
    className="action"
  >
    <button
      className="zippy-btn btn btn-primary"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Add Contact
    </button>
  </div>,
]
`;
